### ISSUE TITLE
Move to archive (automation)

### ISSUE THREAD

**Issue by serrq** [bugsdev2/bookaccio#8]
All those books that I began and after leaved, I can move them to archive for maybe a future re-consideration.

**Comment by bugsdev2** [comment]
Will implement this.

**Comment by bugsdev2** [comment]
As in, inside the book entry, right? Like maybe above the rating stars or someplace like that?

**Comment by bugsdev2** [comment]
I was thinking along the lines of if you move a book to done and the current page doesn't match with the total no. of pages (meaning, you haven't finished the book), it'll show the label "not finished" automatically in the done page.

**Comment by bugsdev2** [comment]
Yes. If the book has 200 pages and if you stop at 199 and move to done, it will show as not finished.

**Comment by bugsdev2** [comment]
Yes yes. You can edit pages read while in the done section as well. 
So, once the pages read match the total number of pages, the label will be removed.

**Comment by bugsdev2** [comment]
> So when I reach 197/200 and I finished the book, simply I go to the book editor and turn 197 in 200/200 to get a label match?
> 
> 100% is "finished" (label) match.
> 
> Do I understand properly your idea?

Yes. So, if your book is in the 'Done' section and the current page isn't the same as the total no. of pages (197/200), it will show an 'Unfinished' tag. When you update the book in the done section (200/200), it will automatically remove the unfinished tag. That's how it i going to work.


> Of course just we add the "finished" label when the 100% of reading is reached.
> 
> There is no reason to add a "not finished" label I believe.

I'm working with the assumption that we usually finish the books that we started and therefore will have more of 'finished' books than unfinished ones. 
So, the finished books will have no tags. But the unfinished ones will have an 'unfinished' tag.

Hope I've made it clear.

> Will be "not finished" books shown (grouped) on top of the list (in Done tab)? Will be there preferences about the sorting and grouping behaviors?
> 
> Just to have an idea of app behavior.

Currently, the sorting is limited in the sense that you can sort either by title or by the start date. Within start date, you can sort by earliest to latest or latest to earliest.
So, in short. No. The tags won't be able to be grouped together on top or bottom. 

Here's a small preview of how it would look like:


![photo_2025-01-08_06-45-19](https://github.com/user-attachments/assets/2db3100d-5e8b-4571-bf98-5fb58962dbb1)

**Comment by bugsdev2** [comment]
> Just a curiosity.
> 
> My ebook reader app is Librera (fdroid and github).
> 
> Has Librera internal APIs in order to communicate with Bookaccio?

I do not think Librera has an api and stuff. At least, my searches didn't yield any results. 
If you or anyone else do know of it/come across it, do share. I'll look into it and see if anything can be done.

**Comment by bugsdev2** [comment]
If this has been resolved, shall I mark this as closed?
